src/components/FeaturePanel/PublicTransport/helpers.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * Sorts an array of items based on their order in a reference array.
+ *
+ * @template TInput The type of the items in the unsorted array.
+ * @template TRef The type of the items in the reference array.
+ * @param {TInput[]} unsorted - The array of items to sort.
+ * @param {TRef[]} reference - The reference array defining the desired order.
+ * @param {(item: TInput) => TRef} functor - A function that maps each item in the unsorted array to a corresponding item in the reference array.
+ * @returns {TInput[]} The sorted array.
+ *
+ * @example
+ * ```typescript
+ * sortBy(
+ * 	[{ name: "cherry" }, { name: "apple" }, { name: "banana" }, { name: "date" }],
+ * 	["apple", "banana", "cherry"],
+ * 	(item) => item.name
+ * );
+ * ```
+ */
+export function sortBy<TInput, TRef>(
```

Comment:
What is the difference between this and lodash sortBy ?

Reply:
I named that function very bad. My sortBy is explained in the js-doc comment and the unit test.  
[Here](https://lodash.com/docs#sortBy) is a link to `sortBy` as it is provided by lodash

Reply:
I named it `sortByReference` this way it should be clear what the function does